lib/galaxy/tools/deps/__init__.py

```diff
@@ -107,10 +108,17 @@ def uses_tool_shed_dependencies(self):
 
     def find_dep( self, name, version=None, type='package', **kwds ):
         log.debug('Find dependency %s version %s' % (name, version))
-        for resolver in self.dependency_resolvers:
+        index = kwds.get('index', None)
+        require_exact = kwds.get('exact', False)
+        for i, resolver in enumerate(self.dependency_resolvers):
+            if index is not None and i != index:
+                continue
+
             dependency = resolver.resolve( name, version, type, **kwds )
             log.debug('Resolver %s returned %s (isnull? %s)' % (resolver.resolver_type, dependency,
                                                                 dependency == INDETERMINATE_DEPENDENCY))
+            if require_exact and not dependency.exact:
+                dependency = INDETERMINATE_DEPENDENCY
```

Comment:
No, this looks to be what I meant. I want to check that second if condition, regardless of the first condition.